frontends/infinite-corridor/src/components/LearningResourceCard.tsx

```diff
@@ -30,7 +31,7 @@ const LearningResourceCard: React.FC<LearningResourceCardProps> = ({
   const activateResource = useActivateResourceDrawer()
   const addToList = useAddToListDialog()
   const { user } = SETTINGS
-  const isInList = resource.lists.length > 0 || resource.is_favorite
+  const isInList = (resource.lists?.length ?? 0) > 0 || resource.is_favorite
```

Comment:
This is the runtime code change. Other changes are testing and fixing the typedef for search results.